bionic/cache.py

```diff
@@ -961,20 +961,33 @@ def dependencies_exactly_match(self, prov):
 
 
 # Helpers for managing files.
-FILE_URL_PREFIX = 'file://'
+FILE_SCHEME = 'file'
+GCS_SCHEME = 'gs'
+SUPPORTED_SCHEMES = [FILE_SCHEME, GCS_SCHEME]
 
 
 def is_file_url(url):
-    return url.startswith(FILE_URL_PREFIX)
+    result = urlparse(url)
+    return result.scheme == FILE_SCHEME
+
+
+def is_gcs_url(url):
+    result = urlparse(url)
+    return result.scheme == GCS_SCHEME
+
+
+def is_absolute_url(url):
```

Comment:
If we see an unrecognized scheme here (like `s3:`), this will return `False`, right? Is that what we want?

Reply:
ah right, I don't think we want that. How do you feel about throwing an error here if it's not in `SUPPORTED_SCHEMES` list?

Reply:
That sounds reasonable to me, since the set of schemes in use is under our control.